R/convert-from-datacamp.R

```diff
@@ -1,44 +1,86 @@
-
-datacamp_to_learnr <- function(.repo) {
+#' datacamp_to_learnr
+#'
+#'
+#' @export
+datacamp_to_learnr <- function(cur_path=NULL, rename=NULL, author_name="Default", has_data=T) {
   # TODO: Check if working dir is a repo
   # TODO: Check if any uncommitted files are present
   # TODO: Check that the necessary files exist
   # TODO: Check if a "original materials" git branch exists
 
-# options(usethis.quiet = TRUE)
+options(usethis.quiet = TRUE)
+
+if(is.null(cur_path)){
+  cur_path=getwd()
+}
 #
 # # For accidental sourcing... -_-
 # stop()
 #
 # # Checking if project is under git and if changes are not committed
-# if (git2r::in_repository())
-#     stop("Please make that the files are under Git version control.")
-# if (!is.null(git2r::status()$unstaged))
-#     stop("Please make sure to commit changes before running.")
+if (!in_repository(cur_path))
+    stop("Please make that the files are under Git version control.")
+if (length(status(cur_path)$unstaged)!=0)
+    stop("Please make sure to commit changes before running.")
 #
 # # Listing all files -------------------------------------------------------
-#
-# all_files <- dir_ls(here(), recursive = TRUE, type = "file", all = TRUE)
-# proj_name <- str_subset(all_files, "\\.Rproj$")
-#
+
+## needs to check to see if there are dashes in the folder name
+
+pkgname <- tail(strsplit(cur_path, "/")[[1]], 1)
+if(grepl("-", pkgname)){
+  if(is.null(rename)){
+    pkgname <- gsub("-", "", pkgname)
+  } else {
+    pkgname <- rename
+  }
+
+  new_path <- strsplit(cur_path, "/")[[1]]
+  new_path <- new_path[-length(new_path)]
+  # new_path <- new_path[-1]
+  new_path <- paste(new_path, collapse=.Platform$file.sep)
+  new_path <- file.path(new_path, pkgname)
+  file.rename(cur_path, new_path)
+  cur_path <- new_path
+}
+setwd(cur_path)
+pkgname <- paste(pkgname, ".Rproj", sep="")
+
+all_files <- dir_ls(cur_path, recurse = TRUE, type = "file", all = TRUE)
+proj_name <- str_subset(all_files, "\\.Rproj$")
+
+
+
+#### No RPoject, which would seem to be default??
+if(length(proj_name)==0){
+  tmp <- paste(pkgname, ".Rproj", sep="")
+  proj_name = file.path(cur_path, tmp)
+}
 # # Create as package -------------------------------------------------------
-#
-# create_package(path_dir(proj_name))
-# use_data_raw()
 
-# chapters <- str_subset(all_files, "chapter.\\.md")
 
-# new_chapter_path <- here("inst/tutorials", path_file(chapters)) %>%
-#     str_replace("\\.md", ".Rmd")
-#
-# file_move(chapters, new_chapter_path)
-#
+create_package(path_dir(proj_name))
+if(has_data){
+  use_data_raw(open=F)
+}
+
+chapters <- str_subset(all_files, "chapter.\\.md")
+
+new_chapter_path <- file.path(cur_path, "inst/tutorials/", path_file(chapters)) %>%
```

Comment:
there's actually a function to create tutorials: `usethis::use_tutorial()`. Do you want to try switching the code over to that?

Reply:
Looking at the `use_tutorial()` documentation it seems like that is for creating blank tutorials whereas here we are moving the datacamp chapters into the tutorial folder so they can be edited into the learnr format.

Reply:
That's fine. We can replace over the created template tutorial. `use_tutorial()` also adds several other things, such as addition to `.Rbuildignore` etc.

Reply:
Set it so that it now uses use_tutorial() and then overwrites it with the right files.